contracts/example/src/contract.rs

```diff
@@ -36,34 +40,83 @@ impl AxelarExecutableInterface for Example {
     ) -> Result<(), ExampleError> {
         Self::validate_message(&env, &source_chain, &message_id, &source_address, &payload)?;
 
-        event::executed(&env, source_chain, message_id, source_address, payload);
+        ExecutedEvent {
+            source_chain,
+            message_id,
+            source_address,
+            payload,
+        }
+        .emit(&env);
+
         Ok(())
     }
 }
 
+#[contractimpl]
+impl InterchainTokenExecutableInterface for Example {
+    fn interchain_token_service(env: &Env) -> Address {
+        env.storage()
+            .instance()
+            .get(&DataKey::InterchainTokenService)
+            .expect("ITS not found")
+    }
+
+    fn execute_with_interchain_token(
+        env: &Env,
+        source_chain: String,
+        message_id: String,
+        source_address: Bytes,
+        payload: Bytes,
+        token_id: BytesN<32>,
+        token_address: Address,
+        amount: i128,
+    ) {
+        Self::validate(env);
+
+        TokenReceivedEvent {
+            source_chain,
+            message_id,
+            source_address,
+            payload,
```

Comment:
To make this a bit more interesting, decode the payload as an `Address::from_string_bytes` and forward the token to that address. Basically this allows sending a cross-chain token but through the executable instead of ITS directly to show case the programmability